modules/download/DownloadQuery.js

```diff
@@ -154,9 +154,7 @@ class DownloadQuery extends Query {
             args.push("--sponsorblock-remove");
             args.push(this.environment.settings.sponsorblockRemove);
         }
-        //if(this.environment.settings.keepUnmerged) args.push('--keep-video');
-
-        args.push('--keep-video');
+        if(this.environment.settings.keepUnmerged || this.environment.settings.avoidFailingToSaveDuplicateFileName) args.push('--keep-video');
```

Comment:
Hey, KeepUnmerged is now handled at line 217 in the old code and line 215 in this change.